src/main/java/org/openrewrite/java/logging/ChangeLombokLogAnnotation.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.logging;
+
+import lombok.AllArgsConstructor;
+import org.openrewrite.Option;
+import org.openrewrite.Recipe;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.ChangeType;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@AllArgsConstructor
+public class ChangeLombokLogAnnotation extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Replace any Lombok log annotations with target logging framework annotation";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replace Lombok annotations such as `@CommonsLog` and `@Log4j` with the target logging framework annotation, or `@Sl4fj` if not provided.";
+    }
+
+    @Option(displayName = "Logging framework",
+            description = "The logging framework to use.",
+            valid = {"SLF4J", "Log4J1", "Log4J2", "JUL", "COMMONS"},
+            required = false)
+    @Nullable
+    private String loggingFramework;
```

Comment:
Should we have an enum for this?

Reply:
We do; but it's not uses as an option yet, possibly because of the case insensitive matching applied here
https://github.com/openrewrite/rewrite-logging-frameworks/blob/f3396b04988af0bde3656bda05a430fce1f38b81/src/main/java/org/openrewrite/java/logging/LoggingFramework.java#L46-L54

Reply:
For now I'd copied the approach used in other recipes here, just to remain on the safe side 
https://github.com/openrewrite/rewrite-logging-frameworks/blob/f3396b04988af0bde3656bda05a430fce1f38b81/src/main/java/org/openrewrite/java/logging/SystemOutToLogging.java#L81
https://github.com/openrewrite/rewrite-logging-frameworks/blob/f3396b04988af0bde3656bda05a430fce1f38b81/src/main/java/org/openrewrite/java/logging/SystemErrToLogging.java#L79
https://github.com/openrewrite/rewrite-logging-frameworks/blob/f3396b04988af0bde3656bda05a430fce1f38b81/src/main/java/org/openrewrite/java/logging/PrintStackTraceToLogError.java#L76
https://github.com/openrewrite/rewrite-logging-frameworks/blob/f3396b04988af0bde3656bda05a430fce1f38b81/src/main/java/org/openrewrite/java/logging/SystemPrintToLogging.java#L44-L49

Reply:
Looks like [we might already load enum values case insensitively](https://github.com/openrewrite/rewrite/blob/eee01ddc30f343f98c355a8c9344081c228c692e/rewrite-core/src/main/java/org/openrewrite/config/YamlResourceLoader.java#L142); I'll followup this change in a second PR just to make sure that's covered.

Reply:
Tried this on a new branch briefly, but then we run into issues with not being able to set a default value for the `LoggingFramework` :/, unless I overlooked something. I've got the changes stashed locally, but don't think there's an easy way to make this work. Did encounter some other clean ups we can do which I'll apply separately.